recipes/xerces-c/all/conanfile.py

```diff
@@ -49,41 +100,25 @@ def configure(self):
             del self.options.fPIC
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
 
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self)
         # https://xerces.apache.org/xerces-c/build-3.html
-        self._cmake.definitions["network-accessor"] = {
-            "Windows": "winsock",
-            "Macos": "cfurl",
-            "Linux": "socket",
-        }.get(str(self.settings.os))
-        self._cmake.definitions["transcoder"] = {
-            "Windows": "windows",
-            "Macos": "macosunicodeconverter",
-            "Linux": "gnuiconv",
-        }.get(str(self.settings.os))
+        self._cmake.definitions["network-accessor"] = self.options.network_accessor
+        self._cmake.definitions["transcoder"] = self.options.transcoder
         self._cmake.definitions["message-loader"] = "inmemory"
-
         self._cmake.definitions["xmlch-type"] = self.options.char_type
-
-        self._cmake.definitions["mutex-manager"] = {
-            "Windows": "windows",
-            "Macos": "posix",
-            "Linux": "posix",
-        }.get(str(self.settings.os))
-        # avoid picking up system dependency
-        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_CURL"] = True
-        self._cmake.definitions["CMAKE_DISABLE_FIND_PACKAGE_ICU"] = True
```

Comment:
What happen now when libcurl and icu are not in the dependency graph? Seems to not be a good idea to completly remove this block.

Reply:
Seems like the system ICU and curl are picked up in this case:
```log
-- Found the following ICU libraries:
--   uc (required)
--   data (required)
-- Found ICU: /usr/include (found version "69.1") 

...

-- Found CURL: /usr/lib/libcurl.so (found version "7.78.0")
```
Since this could lead to problems when either is not installed I also think it's best to not remove this block